### DISCUSSION TITLE
Issues with adding Luna keyboard

### DISCUSSION THREAD

**Discussion by mattdibi** [manna-harbour/miryoku#183] [Q&A] [answered]
Hi there 👋 

I'm currently trying to add the [Luna keyboard by Mindhatch](https://kbd.news/Luna-keyboard-1820.html) to be supported by Miryoku ZMK (and if you're interested I might be able to create a PR to the main repo).

Currently I built the left half and everything is working (home row mods, thumb combos, all the base layer keys) **except for the top/bottom row combos**. I'm almost sure this is a user error but I cannot figure out what I did wrong.

[Here you can find my fork](https://github.com/mattdibi/miryoku_zmk/tree/mattdibi/luna_support) with the changes.

Until now I:
- Added the `.github/workflows/outboards/shields/luna` file referring to [Mindhatch's ZMK config](https://github.com/mindhatch/zmk-config-luna)
- Added the `config/luna.keymap` which simply re-uses the hummingbird keymap
- Added the `config/luna.conf` for setting `CONFIG_ZMK_COMBO_MAX_COMBOS_PER_KEY=16`

Finally I set the bottom row combos term to 200ms to make it easier to test.

This is my build configuration:
![immagine](https://user-images.githubusercontent.com/22748355/210545023-edd0ffee-bf95-4e3f-8591-6e6245d414fb.png)

I performed quite a few tests with building but cannot figure out the issue

Does somebody spot something obvious that I'm missing?

**Comment by manna-harbour** [comment, marked as answer]
What you have is correct. The issue is this https://github.com/manna-harbour/miryoku/issues/56:

>Add hand written bottom row combos to Miryoku ZMK supporting default layout options

So, currently only the default (Colemak Mod-DH) is supported, and QWERTY is not supported.

See https://github.com/manna-harbour/miryoku_zmk/pull/8.

I'll add QWERTY support to the main branch now, so look out for that commit.

After that, a PR would be great! Just add the copyright header to the outboard file, and add an entry to https://github.com/manna-harbour/miryoku_zmk/blob/master/.github/workflows/test-all-promicro-shields.yml.

**Reply by mattdibi** [reply]
... aaaaaaaaaaaaand it works! 🥳 

Thank you very much for the support 😃 ! I'll open the PR ASAP :)
